.github/workflows/python_release.yml

```diff
@@ -0,0 +1,115 @@
+name: Release
+
+on:
+  push:
+    tags: [ 'python-v*' ]
+
+defaults:
+  run:
+    working-directory: ./python
+
+jobs:
+  validate-release-tag:
+    name: Validate git tag
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          PUSHED_TAG=${GITHUB_REF##*/}
+          CURR_VER=$( grep version Cargo.toml | head -n 1 | awk '{print $3}' | tr -d '"' )
+          if [[ "${PUSHED_TAG}" != "python-v${CURR_VER}" ]]; then
+            echo "Cargo metadata has version set to ${CURR_VER}, but got pushed tag ${PUSHED_TAG}."
+            exit 1
+          fi
+
+  release-github-pypi:
+    needs: validate-release-tag
+    name: PyPI release
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macOS-10.15
+          - macOS-11.0
+          - windows-2019
+        include:
+          - target: x86_64-apple-darwin
+            os: macOS-10.15
+          - target: x86_64-apple-darwin
+            os: macOS-11.0
+          - target: x86_64-pc-windows-msvc
+            os: windows-2019
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          target: ${{ matrix.target }}
+
+      - name: Install matruin
+        run: |
+          cargo install --git https://github.com/PyO3/maturin.git --rev 98636cea89c328b3eba4ebb548124f75c8018200 maturin
+
+      - uses: actions/setup-python@v2
```

Comment:
I'm assuming that by setting up all these Python point releases, a binary wheel will be pushed for each point release?

Reply:
that's right, maturin will pick up all available python versions from PATH